src/System.CommandLine/ArgumentResult.cs

```diff
@@ -5,14 +5,19 @@ namespace System.CommandLine
 {
     public abstract class ArgumentResult
     {
-        private protected ArgumentResult()
+        protected internal ArgumentResult(IArgument argument)
```

Comment:
Given that this is an abstract class, can't this simply be `protected`

Reply:
Changing back to `private protected` for the moment. Ultimately I think this class needs to be rewritten so that it inherits `SymbolResult`.